mono/mini/debugger-agent.c

```diff
@@ -4759,7 +4759,17 @@ get_this (StackFrame *frame)
 
 	guint8 * addr = (guint8 *)mono_arch_context_get_int_reg (&frame->ctx, var->index & ~MONO_DEBUG_VAR_ADDRESS_MODE_FLAGS);
 	addr += (gint32)var->offset;
-	return *(MonoObject**)addr;
+
+	if (frame->method->klass->valuetype) {
+		MonoError error;
+		/* Roslyn generates valuetypes in optimize mode */
+		/* This is ok to do because the caller will not modify the object */
+		MonoObject *obj = mono_value_box_checked (frame->domain, frame->method->klass, *(gpointer*)addr, &error);
```

Comment:
So `addr` is a pointer to a pointer to the value type, not a pointer to the value type?

Reply:
It might be wrong, the test suite doesn't hit the code path which actually uses the object, and it just uses the object as an id, so this fix is probably wrong.